### PR TITLE
Just some needed backend changes

### PR DESCRIPTION
**Pull request by Kiwi-x-Kiwi**
- Prevented duplicate entries in some models (Like, Review, Rating, etc...)
- Added routes for Rating and Mechanic (Working in Postman)

### CODE REVIEW

File: server/routes/games.js [hunk 1]
```diff
@@ -36,7 +38,7 @@ router.get('/search', async (req, res, next) => {
 
     Game.find({ $text: { $search: terms } })
         .limit(limit)
-        .populate('likes')
+        .populate(populatable_virtuals)
```
Comment (AndresMWeber) [inline server/routes/games.js:41]: I did all this =X

File: server/models/Like.js [hunk 2]
```diff
@@ -19,6 +19,8 @@ const likeSchema = new Schema({
     },
 })
 
+likeSchema.index({ game: 1, user: 1 }, { unique: true })
```
Comment (AndresMWeber) [inline server/models/Like.js:22]: 👍

File: server/models/Game.js [hunk 3]
```diff
@@ -30,13 +35,29 @@ const gameSchema = new mongoose.Schema({
     // }]
```
Comment (AndresMWeber) [inline server/models/Game.js:35]: The categories thing is breaking my display...we'll have to talk about how to fix this or populate my mongoose properly soon.

**Review by AndresMWeber — APPROVED** [review]

File: server/models/Game.js (hunk 3 above)
Comment (Kiwi-x-Kiwi) [reply server/models/Game.js:35]: Ok.  I think I ran into some issues too but then resolved it.  Not sure what the issue might be.

**Pull request merged**